feature/mtu/largeippacket/otg_tests/large_ip_packet_transmission/large_ip_packet_transmission_test.go

```diff
@@ -0,0 +1,723 @@
+	gnmi.Replace(t, dut, lacpPath.Config(), lacp)
+
+	agg := &oc.Interface{
+		Name: ygot.String(bundleID),
+		Mtu:  ygot.Uint16(mtu),
+		Type: oc.IETFInterfaces_InterfaceType_ieee8023adLag,
+	}
+	if !deviations.OmitL2MTU(dut) {
+		agg.Mtu = ygot.Uint16(mtu + 14)
+	}
+	agg.Description = ygot.String(fmt.Sprintf("dutLag-%s", bundleID))
+	if deviations.InterfaceEnabled(dut) {
+		agg.Enabled = ygot.Bool(true)
+	}
+
+	subInterface := agg.GetOrCreateSubinterface(subInterfaceIndex)
+	v4SubInterface := subInterface.GetOrCreateIpv4()
+	v4SubInterface.SetMtu(mtu)
+	if deviations.InterfaceEnabled(dut) {
+		v4SubInterface.Enabled = ygot.Bool(true)
+	}
+	v4Address := v4SubInterface.GetOrCreateAddress(lag.IPv4)
+	v4Address.PrefixLength = ygot.Uint8(ipv4PrefixLen)
+
+	v6SubInterface := subInterface.GetOrCreateIpv6()
+	v6SubInterface.SetMtu(mtu)
+	if deviations.InterfaceEnabled(dut) {
+		v6SubInterface.Enabled = ygot.Bool(true)
+	}
+	v6Address := v6SubInterface.GetOrCreateAddress(lag.IPv6)
+	v6Address.PrefixLength = ygot.Uint8(ipv6PrefixLen)
+
+	intfAgg := agg.GetOrCreateAggregation()
+	intfAgg.LagType = oc.IfAggregate_AggregationType_STATIC
+
+	aggPath := gnmiOCRoot.Interface(bundleID)
+	gnmi.Replace(t, dut, aggPath.Config(), agg)
+
+	if deviations.ExplicitInterfaceInDefaultVRF(dut) {
+		fptest.AssignToNetworkInstance(
+			t, dut, bundleID, deviations.DefaultNetworkInstance(dut), subInterfaceIndex,
+		)
+	}
+
+	// if we didnt setup the ports in the lag before
+	if !deviations.AggregateAtomicUpdate(dut) {
+		for _, port := range bundleMembers {
+			intf := &oc.Interface{Name: ygot.String(port.Name())}
+			intf.GetOrCreateEthernet().AggregateId = ygot.String(bundleID)
+			intf.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+
+			if deviations.InterfaceEnabled(dut) {
+				intf.Enabled = ygot.Bool(true)
+			}
+
+			if deviations.ExplicitPortSpeed(dut) {
+				fptest.SetPortSpeed(t, port)
+			}
+
+			intfPath := gnmiOCRoot.Interface(port.Name())
+
+			gnmi.Replace(t, dut, intfPath.Config(), intf)
+		}
+	}
+
+	verifyDUTPort(t, dut, *agg.Name)
+
+	return bundleID
+}
+
+func configureATEBundles(
+	allAtePorts []*ondatra.Port,
+	bundleMemberCount int,
+) gosnappi.Config {
+	otgConfig := gosnappi.NewConfig()
+
+	otgConfig = configureATEBundle(
+		otgConfig,
+		atePort1,
+		dutPort1,
+		allAtePorts[0:bundleMemberCount],
+		1,
+	)
+	otgConfig = configureATEBundle(
+		otgConfig,
+		atePort2,
+		dutPort2,
+		allAtePorts[bundleMemberCount:2*bundleMemberCount],
+		2,
+	)
+
+	portNames := make([]string, len(allAtePorts))
+	for idx, port := range allAtePorts {
+		portNames[idx] = port.ID()
+	}
+
+	// note that it seems max in otg containers is 9000 so bundle tests > 1500 bytes will fail,
+	// for whatever reason individual ports work just fine > 1500 bytes though! also, physical gear
+	// seems to work just fine as well, so we'll set this to the max we can for kne tests.
+	otgConfig.Layer1().Add().SetName("layerOne").SetPortNames(portNames).SetMtu(9000)
```

Comment:
OTG `Layer1` objects default the port speed to 10 gpbs, which is causing issues for this test. Can we explicitly set the speed for the ports here based on the information in the reservation? (via something like `SetSpeed(Layer1Speed.SPEED_100_GBPS)`)

Reply:
@jasdeep-hundal copy -- updated to set based on zeroith port speed in the testbed ports. perhaps not the absolute best but looks like a reasonable approach, lemme know if you wanna see any tweaks!